### PR TITLE
Update assistant to not crash on startup.

### PR DESCRIPTION
**Pull request by stnguyen90**
## What does this PR do?

If the _APP_ASSISTANT_OPENAI_API_KEY env var was not set, the assistant would crash. This prevents that from happening and returns an error message if the key is not set.

### With valid _APP_ASSISTANT_OPENAI_API_KEY set

Logs:

image

Request:

image

### With invalid _APP_ASSISTANT_OPENAI_API_KEY set

Logs:

image

...

image

Request:

image

### Without _APP_ASSISTANT_OPENAI_API_KEY set

Logs:

image

Request:

image

## Related PRs and Issues

* https://github.com/appwrite/appwrite/issues/6217

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -30,13 +36,19 @@ app.post("/", async (req, res) => {
...
     question: templated,
   });
 
   res.end();
 });
 
-app.listen(port, () => {
+app.listen(port, async () => {
   console.log(`Started server on port: ${port}`);
+  console.log('initializing search index...');
```
Comment (TGlide) [inline main.js:48]:
```suggestion
  console.log('Initializing search index...');
```

Comment (TGlide) [reply main.js:48]: We should also probably log when it has been initialized

**Review by TGlide — APPROVED** [review]

File: main.js (hunk 1 above)
Comment (stnguyen90) [reply main.js:48]: updated!

**Review by TGlide — APPROVED** [review]

File: main.js [hunk 2]
```diff
@@ -30,13 +36,20 @@ app.post("/", async (req, res) => {
...
```
Comment (TorstenDittmann) [inline main.js:48]: I think still a typo with lowercase `i`

**Review by TorstenDittmann — APPROVED** [review]

File: main.js (hunk 2 above)
Comment (stnguyen90) [reply main.js]: Updated!

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**